subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/SmokeContinuousIntegrationTest.groovy

```diff
@@ -32,25 +32,85 @@ class SmokeContinuousIntegrationTest extends AbstractContinuousIntegrationTest {
         }
     }
 
+    def "detects changes when no files are in the project"() {
+        given:
+        def markerFile = file("input/marker")
+        buildFile << """
+            task build {
+              def inputFile = file("input/marker")
+              inputs.files inputFile
+              outputs.files "build/marker"
+              doLast {
+                println "exists: " + inputFile.exists()
+              }
+            }
+        """
+
+        when:
+        succeeds("build")
+        then:
+        output.contains "exists: false"
+
+        when:
+        waitBeforeModification(markerFile)
+        markerFile.text = "created"
+        then:
+        succeeds()
+        output.contains "exists: true"
+    }
+
+    def "detects changes when no snapshotting happens (#description)"() {
+        given:
+        def markerFile = file("input/marker")
+        buildFile << """
+            task build {
+              def inputFile = file("input/marker")
+              inputs.files inputFile
+              doLast {
+                println "value: " + inputFile.text
+              }
+            }
+        """
+
+        when:
+        markerFile.text = "original"
+        succeeds("build")
+        then:
+        output.contains "value: original"
+
+        when:
+        waitBeforeModification(markerFile)
+        markerFile.text = "changed"
+        then:
+        succeeds()
+        output.contains "value: changed"
+
+        where:
+        description      | taskConfiguration
+        "no outputs"     | ""
+        "untracked task" | "outputs.file('output/marker'); doNotTrackState('for test')"
+    }
+
     @ToBeFixedForConfigurationCache
     def "basic smoke test"() {
         given:
-        def markerFile = file("marker")
+        def markerFile = file("input/marker")
```

Comment:
This is so that the inputs/outputs are not in the project directory? My understanding is that this change is not really required for this test, but you are doing it for consistency's sake, correct?

Reply:
For Linux, we are now also watching the project root directory. So we'd always get notifications there, even if it wouldn't be watched because of the state of the VFS. That is why I moved all the source inputs into a directory inside of the root project directory.